meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc

```diff
@@ -9,6 +9,9 @@ DEPLOYDIR = "${WORKDIR}/deploy-${PN}"
 
 MENDER_UBOOT_AUTO_CONFIGURE ??= "${@bb.utils.contains('DISTRO_FEATURES', 'mender-uboot', '1', '0', d)}"
 
+MENDER_UBOOT_CONFIG_SYS_MMC_ENV_PART ??= "0"
+MENDER_RESERVED_SPACE_BOOTLOADER_DATA_CHECK ??= "${MENDER_RESERVED_SPACE_BOOTLOADER_DATA}"
```

Comment:
I do not think this is needed, you should be able to override `MENDER_RESERVED_SPACE_BOOTLOADER_DATA` as it is only a weak default defined [here](https://github.com/mendersoftware/meta-mender/blob/1b8af154b547c4fa85538bb321da21eecb36a5b5/meta-mender-core/classes/mender-setup.bbclass#L143), or did this not work for you?

Reply:
@mirzak thanks for the comment.

> I do not think this is needed, you should be able to override MENDER_RESERVED_SPACE_BOOTLOADER_DATA as it is only a weak default defined here,

I did override [MENDER_RESERVED_SPACE_BOOTLOADER_DATA] to 0)[https://github.com/Trellis-Logic/meta-mender-community/pull/1/files#diff-8d248b04423a467e816423303d7b1468R26] to avoid including this in [MENDER_CALC_ROOTFS_SIZE](https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/classes/mender-setup-install.inc#L43).  Maybe it would be easier to weakly define MENDER_CALC_ROOTFS_SIZE instead and override?  If I don't somehow change MENDER_CALC_ROOTFS_SIZE I think this will be the size used in [IMG_CMD_mender](https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/classes/mender-artifactimg.bbclass#L29), but if I set it to 0, which is what I need it to be for rootfs calculations, the checks will fail. Let me know if I'm missing something obvious here.

Reply:
I might be misunderstanding the intent of `MENDER_RESERVED_SPACE_BOOTLOADER_DATA_CHECK` but to me it only looks like you add this variable to be able to override it and this why my comment said why not simply use `MENDER_RESERVED_SPACE_BOOTLOADER_DATA` but I might be missing context. How is this used in your meta-mender-tegra configuration?

Might be easier to view if you clean up the PR and submit it.

Reply:
There are two uses for MENDER_RESERVED_SPACE_BOOTLOADER_DATA today in meta-mender, as I understand it.

1) Calculating rootfs size with [MENDER_CALC_ROOTFS_SIZE](https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/classes/mender-setup-install.inc#L43).
2) [Sanity checking u-boot variables like BOOTENV_SIZE](https://github.com/mendersoftware/meta-mender/blob/1a3a805bb441d38e029a9697cab51e5535a52caf/meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc#L49).

In my case I will be placing the uboot environment in a different MMC partition, so I don't want it to be a part of the rootfs size calculation.

This PR proposes allowing a split of the two, so I can [set MENDER_RESERVED_SPACE_BOOTLOADER_DATA  to 0](https://github.com/Trellis-Logic/meta-mender-community/pull/1/files#diff-5f290d56d86e5e767ae7c7315d99f9b0R11) for the rootfs size calculation but set to the [actual reserved space value](https://github.com/Trellis-Logic/meta-mender-community/pull/1/files#diff-5f290d56d86e5e767ae7c7315d99f9b0R11) for the purposes of sanity checks.

> Might be easier to view if you clean up the PR and submit it.

Will do that tonight.

Reply:
Good description, put that in the commit message :)